libraries/AP_RangeFinder/AP_RangeFinder_BLPing.h

```diff
@@ -0,0 +1,75 @@
+#pragma once
+
+#include "RangeFinder.h"
+#include "RangeFinder_Backend.h"
+
+class AP_RangeFinder_BLPing : public AP_RangeFinder_Backend
+{
+
+public:
+
+    // constructor
+    AP_RangeFinder_BLPing(RangeFinder::RangeFinder_State &_state,
+                            AP_RangeFinder_Params &_params,
+                            AP_SerialManager &serial_manager,
+                            uint8_t serial_instance);
+
+    // static detection function
+    static bool detect(AP_SerialManager &serial_manager, uint8_t serial_instance);
+
+    // update state
+    void update(void) override;
+
+protected:
+
+    virtual MAV_DISTANCE_SENSOR _get_mav_distance_sensor_type() const override {
+        return MAV_DISTANCE_SENSOR_ULTRASOUND;
+    }
+
+private:
+
+    // initialise sensor
+    void init_sensor();
+
+    // send message to sensor
+    void send_message(uint16_t msgid, uint8_t *payload, uint16_t payload_len);
+
+    // read a distance from the sensor
+    bool get_reading(uint16_t &reading_cm);
+
+    // process one byte received on serial port
+    // returns true if a distance message has been successfully parsed
+    // state is stored in msg structure
+    bool parse_byte(uint8_t b);
+
+    enum msg_state_t {
```

Comment:
Make this `enum class` and strip off all of the stuff.

So 
```
enum class msg_state : uint8_t {
   HEADER1,
   HEADER2,
.
.
}
```

and then

```
case msg_state::LEN_L
```

We don't have much precedent for the case of the `enum class`, but I think

```
case MSGState::LEN_L:
```

looks marginally better.

Reply:
as mentioned above I tried to use an enum but ran into troubles in the parser for cases where the message might not be in  the enum